### ISSUE TITLE
export/import courses, chapters, lessons, assignments?

### ISSUE THREAD

**Issue by fgmart-zz** [NatTuck/bottlenose-classic#86]
Seems like a big step forward would be ability to export and import courses and components of courses.

Would be useful in cloning a course, and backing them up / restoring them.

**Comment by NatTuck** [comment]
I was thinking the same thing, there are a couple of different scenarios:
- Course import / export, not including student content. Good for cloning and for sharing courses with other Bottlenose instances.
- "Download a tarball of submissions". Good for offline grading.
- Download all student data (including grades, instructor comments). Good for archival.

It seems like exporting a full course including student data would be good for backups, but I see a couple of problems with that approach:
- Implementing import for it would freeze a lot of application behavior and make development more difficult.
- It would be hard to properly test and the code would probably have silently broken by the time someone needed to do a restore. 
- Using it for backups would require user-side automation and would take a lot of space and bandwidth.

It probably makes more sense to do disaster recovery backups at the system level - e.g. with frequent VM snapshots or a combination of DB dumps and standard file backup tools.

**Comment by marksherman** [comment]
I agree with the first three bullets as useful things. They also happen to
be in descending importance, in my opinion.

I also agree with the last one: at this stage backups are best made at the
system level, as we have full control of that, and it's easy to do. (in
fact, I've already done it)
